homeassistant/components/imgw_pib/binary_sensor.py

```diff
@@ -0,0 +1,97 @@
+"""IMGW-PIB binary sensor platform."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Mapping
+from dataclasses import dataclass
+from typing import Any
+
+from imgw_pib.model import HydrologicalData
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ImgwPibConfigEntry
+from .const import ATTRIBUTION
+from .coordinator import ImgwPibDataUpdateCoordinator
+
+PARALLEL_UPDATES = 1
+
+
+@dataclass(frozen=True, kw_only=True)
+class ImgwPibBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """IMGW-PIB sensor entity description."""
+
+    value: Callable[[HydrologicalData], bool | None]
+    attrs: Callable[[HydrologicalData], dict[str, Any]]
+
+
+BINARY_SENSOR_TYPES: tuple[ImgwPibBinarySensorEntityDescription, ...] = (
+    ImgwPibBinarySensorEntityDescription(
+        key="flood_warning",
+        translation_key="flood_warning",
+        device_class=BinarySensorDeviceClass.SAFETY,
+        value=lambda data: data.flood_warning,
+        attrs=lambda data: {"warning_level": data.flood_warning_level.value},
+    ),
+    ImgwPibBinarySensorEntityDescription(
+        key="flood_alarm",
+        translation_key="flood_alarm",
+        device_class=BinarySensorDeviceClass.SAFETY,
+        value=lambda data: data.flood_alarm,
+        attrs=lambda data: {"alarm_level": data.flood_alarm_level.value},
```

Comment:
What's the attributes for?

Reply:
Attributes show the water level value for which the warning or alert occurs. To be honest, I'm still wondering how these values should be made available to users. As an alternative, these could be `water_level` sensor attributes or separate sensor entities. What do you think?

Reply:
I am assuming these are quite static, I would suggest making them a sensor and making them diagnostic. If they are truly static, consider disabling them by default

Reply:
These values change but very rarely, every few months or even years. In fact, it will be better to add them as disabled by default sensors. Thanks.